### PR TITLE
[aggregator] Evenly space reset data points

### PR DESCRIPTION
**Pull request by wesleyk**
Previously, reset ops would create a new 0 datapoint
exactly 1 second after the actual agg datapoint.
This can create jagged rates at windows close to the
resolution period.

Instead, we space apart the reset datapoints to be
equidistant from one another. We do this by setting
them to be half the resolution period away from the
datapoint.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Comment by ryanhall07** [comment]
might be nice to link to some prom doc/code for why this works

File: src/metrics/transformation/unary_multi.go [hunk 1]
```diff
@@ -41,7 +44,11 @@ import "time"
 // Currently only a single extra datapoint per aggregation is supported. If multiple transforms in an aggregation emit
 // an additional datapoint, only the last one is used.
 func transformReset() UnaryMultiOutputTransform {
-	return UnaryMultiOutputTransformFn(func(dp Datapoint) (Datapoint, Datapoint) {
-		return dp, Datapoint{Value: 0, TimeNanos: dp.TimeNanos + int64(time.Second)}
+	return UnaryMultiOutputTransformFn(func(dp Datapoint, resolution time.Duration) (Datapoint, Datapoint) {
+		// Add the reset datapoint to be half the resolution period to ensure equal spacing between datapoints.
+		// We take the max with 1 to ensure there's at least a 1 second gap.
```
Comment (ryanhall07) [inline src/metrics/transformation/unary_multi.go:49]: this seems unnecessary and would prevent 1s resolutions. why not just support 1s resolutions with 500ms resetwindow?

**Review by ryanhall07 — APPROVED** [review]

**Review by robskillington — APPROVED** [review]
LGTM

**Review by shreyassrivatsan — APPROVED** [review]
🎉

File: src/metrics/transformation/unary_multi.go [hunk 2]
```diff
@@ -41,7 +44,11 @@ import "time"
 // Currently only a single extra datapoint per aggregation is supported. If multiple transforms in an aggregation emit
 // an additional datapoint, only the last one is used.
 func transformReset() UnaryMultiOutputTransform {
-	return UnaryMultiOutputTransformFn(func(dp Datapoint) (Datapoint, Datapoint) {
-		return dp, Datapoint{Value: 0, TimeNanos: dp.TimeNanos + int64(time.Second)}
+	return UnaryMultiOutputTransformFn(func(dp Datapoint, resolution time.Duration) (Datapoint, Datapoint) {
+		// Add the reset datapoint to be half the resolution period to ensure equal spacing between datapoints.
+		// We take the max with 1 to ensure there's at least a 1 nanosecond gap.
```
Comment (ryanhall07) [inline src/metrics/transformation/unary_multi.go:49]: lol nanoscale

**Pull request merged**